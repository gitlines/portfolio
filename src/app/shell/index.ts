export * from './shell-loading/shell-loading.component';
export * from './shell.component';
export * from './shell.module';
