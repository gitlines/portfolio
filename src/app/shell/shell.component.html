<nav class="nav">
   <h1 class="nav__header">
      <a routerLink="/">Portfolio</a>
   </h1>

   <ul class="nav__links">
      <li><a class="nav--link" routerLink="/users" routerLinkActive="nav__link--active">Users</a></li>
   </ul>
</nav>

<router-outlet></router-outlet>
