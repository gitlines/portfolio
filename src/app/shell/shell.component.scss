@import 'palette';
@import 'animation';

:host {
   display: block;
   padding: 1rem;
   padding-top: calc(2rem + 2em);
}

a {
   text-decoration: none;
}

a,
a:visited,
a:hover,
a:active {
   color: inherit;
}

.nav {
   @include theme(dark);

   position: fixed;
   display: flex;
   align-items: center;
   top: 0;
   left: 0;
   right: 0;
   padding: 1rem;
}

.nav__header {
   display: block;
   flex: 1 0 auto;
   margin: 0.25rem 0;
   font-size: 2em;

   & a {
      transition: transition(all);

      &:hover,
      &:focus {
         color: get-color(white);
      }
   }
}

.nav__links {
   display: block;
   flex: 0 1 auto;
   margin: 0;
   padding: 0;
   list-style-type: none;
}

.nav--link {
   display: inline-block;
   padding: 0.25rem 0.5rem;
   border-radius: 0.25rem;
   font-size: 1.6em;
   font-weight: 300;
   transition: transition(all);

   &:hover,
   &:focus {
      color: get-color(white);
   }

   &.nav__link--active {
      background-color: get-color(white, $opacity: 0.2);
      box-shadow: 2px 2px 10px get-color(black, $opacity: 0.2);
   }
}
