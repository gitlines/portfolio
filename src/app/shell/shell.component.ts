import { ChangeDetectionStrategy, Component } from '@angular/core';

/**
 * Component used as app's shell.
 */
@Component({
   selector: 'app-shell',
   templateUrl: './shell.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush
})
export class ShellComponent {
   /**
    * Title of the app
    */
   title = 'Portfolio';

   /**
    * Creates an instance of ShellComponent.
    */
   constructor() {}
}
