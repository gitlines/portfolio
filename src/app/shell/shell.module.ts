import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ShellLoadingComponent } from './shell-loading/shell-loading.component';
import { ShellComponent } from './shell.component';

@NgModule({
   declarations: [ShellComponent, ShellLoadingComponent],
   imports: [CommonModule, RouterModule],
   exports: [ShellComponent, ShellLoadingComponent],
   providers: []
})
export class ShellModule {}
