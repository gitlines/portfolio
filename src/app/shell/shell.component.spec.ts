import { NO_ERRORS_SCHEMA } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { ShellComponent } from './shell.component';

describe('ShellComponent', () => {
   let component: ShellComponent;
   let fixture: ComponentFixture<ShellComponent>;
   let element: any;

   beforeEach(async(() => {
      TestBed.configureTestingModule({
         declarations: [ShellComponent],
         schemas: [NO_ERRORS_SCHEMA]
      }).compileComponents();
   }));

   beforeEach(() => {
      fixture = TestBed.createComponent(ShellComponent);
      component = fixture.componentInstance;
      fixture.detectChanges();
      element = fixture.debugElement.nativeElement;
   });

   it('should create', () => {
      expect(component).toBeTruthy();
   });

   it(`should have as title 'app'`, async(() => {
      expect(component.title).toEqual('Portfolio');
   }));

   it('should render title in a h1 tag', async(() => {
      expect(element.querySelector('h1').textContent).toContain('Welcome to Portfolio!');
   }));

   it('should match the snapshot', async(() => {
      expect(fixture).toMatchSnapshot();
   }));
});
