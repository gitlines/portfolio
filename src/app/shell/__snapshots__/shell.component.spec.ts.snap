// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShellComponent should match the snapshot 1`] = `
<app-shell
  title={[Function String]}
>
  <h1>
    Welcome to Portfolio!
  </h1><router-outlet />
</app-shell>
`;
